lib/capsize/deployer.rb

```diff
@@ -65,23 +74,46 @@ def execute!
       write_deploy
       write_stage
 
-      load_requirements
-      capsize_setup(@stage)
-      set_output
-      status = catch(:abort_called_by_capistrano){
+
+      status = run_in_isolation do
+        load_requirements
+        capsize_setup(@stage)
+        set_output
         Capistrano::Application.invoke("#{@stage.name}")
         after_stage_invokations
         Capistrano::Application.invoke(options[:actions])
-      }
-      close_output
+        close_output
+      end
 
-      status != :capistrano_abort
+      status
 
     rescue Exception => error
       handle_error(error)
       return false
     end
 
+    def run_in_isolation
+      read, write = IO.pipe
+
+      pid = fork do
+        read.close
+        begin
+          result = yield
```

Comment:
I put the `result` local variable back in to help handle errors in a similar way to `execute!` did before.